src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,397 @@
+        rpk_argv.push_back(creds->password);
+        rpk_argv.push_back("--sasl-mechanism");
+        rpk_argv.push_back(creds->mechanism);
+    }
+
+    if (params.logs_since.has_value()) {
+        rpk_argv.push_back("--logs-since");
+        rpk_argv.push_back(params.logs_since.value());
+    }
+
+    if (params.logs_until.has_value()) {
+        rpk_argv.push_back("--logs-until");
+        rpk_argv.push_back(params.logs_until.value());
+    }
+
+    if (params.logs_size_limit.has_value()) {
+        rpk_argv.push_back("--logs-size-limit");
+        rpk_argv.push_back(params.logs_size_limit.value());
+    }
+
+    if (params.metrics_interval.has_value()) {
+        rpk_argv.push_back("--metrics-interval");
+        rpk_argv.push_back(params.metrics_interval.value());
+    }
+
+    return rpk_argv;
+}
+
+std::error_code detail::handle_wait_status(
+  ss::experimental::process::wait_status& wstatus,
+  ss::sstring cmd,
+  ss::sstring filename) {
+    auto* exit_status = std::get_if<ss::experimental::process::wait_exited>(
+      &wstatus);
+    if (exit_status != nullptr) {
+        if (exit_status->exit_code != 0) {
+            vlog(
+              bundle_log.error,
+              "Failed to run {}, exit code {}",
+              cmd,
+              exit_status->exit_code);
+        } else {
+            vlog(
+              bundle_log.debug,
+              "{} successful, debug bundle {}",
+              cmd,
+              filename);
+            return rpk::debug_bundle_error::success;
+        }
+    } else {
+        auto* exit_signal
+          = std::get_if<ss::experimental::process::wait_signaled>(&wstatus);
+        if (exit_signal == nullptr) {
+            vlog(
+              bundle_log.error,
+              "Failed to run {} and the exit "
+              "signal is undefined. Debug bundle "
+              "{}",
+              cmd,
+              filename);
+        } else {
+            vlog(
+              bundle_log.error,
+              "Failed to run {}, process "
+              "terminated with signal {}",
+              cmd,
+              exit_signal->terminating_signal);
+        }
+    }
+
+    return rpk::debug_bundle_error::process_fail;
+}
+
+ss::future<> debug_bundle::run_rpk(
+  const debug_bundle_params params,
+  std::optional<debug_bundle_credentials> creds) {
+    gate_guard guard{_rpk_gate};
+    auto units = co_await _rpk_mu.get_units();
+
+    auto filename = detail::make_bundle_filename(
+      get_write_dir(), default_bundle_name());
+    auto rpk_argv = detail::make_rpk_args(_rpk_cmd, filename, params, creds);
+    try {
+        auto p = co_await ss::experimental::spawn_process(
+          _rpk_cmd, {.argv = std::move(rpk_argv), .env = {_host_path}});
+        _process.emplace(std::move(p));
+    } catch (const std::system_error& ex) {
+        vlog(
+          bundle_log.error,
+          "Process spawn failed: {} - {}",
+          ex.code(),
+          ex.what());
+    }
```

Comment:
I was using the consumer because my experiments failed to get any output from the input_stream from stdout. However, that was a while ago so perhaps things will work now that I've been working with this feature for a while. I'll try this out tomorrow.

Reply:
Sounds good. Let me know if you hit a similar issue. From what I can tell by looking at the implementation is that it should work ok, so I can take a look too if you hit some issues and want to avoid the consumer.

Reply:
This works.